contracts/core/libraries/QVHelper.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.19;
+
+/// @title QV Helper Library
+/// @notice A helper library for Quadratic Voting
+/// @dev Handles the voting of recipients and calculates the payout amount for each recipient
+library QVHelper {
+    /// @notice Error thrown when the number of recipients and amounts are not equal on voting
+    error QVHelper_LengthMissmatch();
+
+    /// @notice Struct that holds the state of the voting
+    /// @param totalVotes The total amount of votes casted
+    /// @param totalVoiceCredits The total amount of voice credits casted
+    /// @param recipientVoiceCredits The voice credits casted for each recipient
+    /// @param recipientVotes The votes casted for each recipient
+    struct VotingState {
+        uint256 totalVotes;
+        uint256 totalVoiceCredits;
+        mapping(address => uint256) recipientVoiceCredits;
+        mapping(address => uint256) recipientVotes;
+    }
+
+    /// @notice Calculate the square root of a number (Babylonian method)
+    /// @param _x The number
+    /// @return _y The square root
+    function _sqrt(uint256 _x) internal pure returns (uint256 _y) {
```

Comment:
Would it make sense to have this in a lib so that other strategies/helpers could use it as well ?
We also have solady which we could use https://github.com/Vectorized/solady/blob/main/src/utils/FixedPointMathLib.sol#L647

Reply:
agree! I will import it from a lib